dask_cuda/explicit_comms/comms.py

```diff
@@ -1,15 +1,21 @@
+# Copyright (c) 2021-2025 NVIDIA CORPORATION.
 import asyncio
 import concurrent.futures
 import contextlib
 import time
 import uuid
+import weakref
 from typing import Any, Dict, Hashable, Iterable, List, Optional
 
 import distributed.comm
+from dask.tokenize import tokenize
 from distributed import Client, Worker, default_client, get_worker
 from distributed.comm.addressing import parse_address, parse_host_port, unparse_address
 
-_default_comms = None
+# Mapping tokenize(client ID, [worker addresses]) to CommsContext
+_comms_cache: weakref.WeakValueDictionary[
+    str, "CommsContext"
+] = weakref.WeakValueDictionary()
```

Comment:
I like the idea of using a `WeakValueDictionary` here.

Reply:
Yeah, the `CommsContext holds a reference to a `Client` so we don't want to be the last one referencing it and keeping it alive.

(There's maybe a discussing to be had about whether CommContext needs to be client-specific, or whether two clients might be able to reuse the same explicit comms. Skimming the code it seems like it might be possible, but I don't want to go down that rabbit hole for something that is probably never going to get used).